netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -240,16 +240,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
       }
     });
     // Bind and start to accept incoming connections.
-    ChannelFuture future = b.bind(address).addListener(new ChannelFutureListener() {
-      @Override
-      public void operationComplete(ChannelFuture f) throws Exception {
-        Instrumented<SocketStats> listenSocket = new ListenSocket(f.channel());
-        listenSockets = ImmutableList.of(listenSocket);
-        channelz.addListenSocket(listenSocket);
-      }
-    });
+    ChannelFuture future = b.bind(address);
     try {
       future.await();
+      Instrumented<SocketStats> listenSocket = new ListenSocket(future.channel());
```

Comment:
you need to check for success first.  future.channel may be null.  See a few lines down.

Reply:
Updated